### DISCUSSION TITLE
How to change from TXgate to RXgate in the sequence

### DISCUSSION THREAD

**Discussion by FeiyangLou** [vnegnev/marcos_extras#7] [Q&A] [answered]
Now the sequence gives a TXgate (low on receive and high on transmit), but my T/R switch design needs a high signal for switching to receive mode. How should I modify the sequence to do that?

**Comment by josalggui** [comment, marked as answer]
The MRIBLANKSEQ class (in mriBlankSeq.py) has the method ttl. With this method you can output a ttl by the Tx or by the Rx gate:

self.ttl(tStart, ttlTime, channel="place here 0 or 1", rewrite=True)

**Reply by FeiyangLou** [reply]
But in the sequence files I can only find the rxGate method, so which part of the sequence controls the TX gate? 
PS. In mriBlankSeq.py, line 220, should it be Tx gate instead of Rx gate?

**Reply by josalggui** [reply]
I use to employ rfRecPulse method to run rf pulses with square shape. This method includes both the Tx gating to enable the RFPA (line 366 of mriBlankSeq.py) and the RF amplitude (line 365 of mriBlankSeq.py).

And yes, you are right, it is Tx gate. Thanks to let me know the typo. I will update in the next commit.

**Reply by josalggui** [reply]
By the way, rxGate method says to the red pitaya that it must acquire data, but it does not turn on any voltage. To turn on a voltage you should use ttl method at the same time as the rxGate method.
